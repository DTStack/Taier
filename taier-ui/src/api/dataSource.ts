/*
 * Licensed to the Apache Software Foundation (ASF) under one
 * or more contributor license agreements.  See the NOTICE file
 * distributed with this work for additional information
 * regarding copyright ownership.  The ASF licenses this file
 * to you under the Apache License, Version 2.0 (the
 * "License"); you may not use this file except in compliance
 * with the License.  You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import { queryParse } from '@/utils';
import Swagger from './dataSources-swagger';
import http from './http';

type APIProps = Record<
	keyof typeof Swagger,
	(params: any, config?: any) => Promise<any>
>;

function mapUrlObjToFuncObj(urlObj: typeof Swagger) {
	const keys = Object.keys(urlObj) as (keyof typeof Swagger)[];
	return keys.reduce((pre, cur) => {
		const item = urlObj[cur];
		pre[cur] = async function (params, config: any = {}) {
			const queryParams: any = queryParse(window.location.hash);
			// 所有的请求接口统一带上header
			if (queryParams.datasource_id) {
				config.headers = new Headers({
					dt_datasource_id: queryParams.datasource_id,
				});
			}
			return await http[item.method as 'get' | 'post' | 'postForm'](
				item.url,
				params,
				config,
			);
		};
		return pre;
	}, {} as APIProps);
}
// function mapUrlObjToStrObj(urlObj) {
//   const Url = {};
//   keys(urlObj).forEach((key) => {
//     const item = urlObj[key];
//     Url[key] = item.url;
//   });
//   return Url;
// }

export const API = mapUrlObjToFuncObj(Object.assign({}, Swagger));
// export const URL = mapUrlObjToStrObj(Object.assign({}, Swagger, Restful));
