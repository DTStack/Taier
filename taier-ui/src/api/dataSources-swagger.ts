/*
 * Licensed to the Apache Software Foundation (ASF) under one
 * or more contributor license agreements.  See the NOTICE file
 * distributed with this work for additional information
 * regarding copyright ownership.  The ASF licenses this file
 * to you under the Apache License, Version 2.0 (the
 * "License"); you may not use this file except in compliance
 * with the License.  You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

//每个模块定义的相关接口
// const api = '/api/publicService';
const api = '/taier/dataSource';

export default {
	getOfflineTableList: {
		method: 'post',
		url: `${api}/addDs/tablelist`,
	},
	getOfflineTableColumn: {
		method: 'post',
		url: `${api}/addDs/tablecolumn`,
	},
	getOfflineColumnForSyncopate: {
		method: 'post',
		url: `${api}/addDs/columnForSyncopate`,
	},
	getHivePartitions: {
		method: 'post',
		url: `${api}/addDs/getHivePartitions`,
	},
	getDataPreview: {
		method: 'post',
		url: `${api}/addDs/preview`,
	},
	getAllSchemas: {
		method: 'post',
		url: `${api}/addDs/getAllSchemas`,
	},
	queryByTenantId: {
		method: 'get',
		url: `${api}/manager/queryByTenantId`,
	},
	//数据源列表分页信息
	dataSourcepage: {
		method: 'post',
		url: `${api}/manager/page`,
	},
	//数据源类型下拉列表
	typeList: {
		method: 'post',
		url: `${api}/manager/type/list`,
	},
	// 授权产品下拉列表
	productList: {
		method: 'post',
		url: `${api}/manager/product/list`,
	},
	//删除一条数据源实例
	dataSourceDelete: {
		method: 'post',
		url: `${api}/manager/delete`,
	},
	//产品授权界面
	authProductList: {
		method: 'post',
		url: `${api}/manager/auth/product/list`,
	},
	// 产品授权
	dataSoProAuth: {
		method: 'post',
		url: `${api}/manager/product/auth`,
	},
	//获取数据源分类类目列表
	queryDsClassifyList: {
		method: 'post',
		url: `${api}/addDs/queryDsClassifyList`,
	},
	//根据分类获取数据源类型
	queryDsTypeByClassify: {
		method: 'post',
		url: `${api}/addDs/queryDsTypeByClassify`,
	},
	//根据数据源类型获取版本列表
	queryDsVersionByType: {
		method: 'post',
		url: `${api}/addDs/queryDsVersionByType`,
	},
	//根据数据库类型和版本查找表单模版
	findTemplateByTypeVersion: {
		method: 'post',
		url: `${api}/dsForm/findFormByTypeVersion`,
	},
	// 获取数据源与租户交集的产品列表
	queryProductList: {
		method: 'post',
		url: `${api}/addDs/queryAppList`,
	},
	// 添加数据源
	addDatasource: {
		method: 'post',
		url: `${api}/addDs/addOrUpdateSource`,
	},
	//上传Kerberos添加数据源
	addOrUpdateSourceWithKerberos: {
		method: 'postForm',
		url: `${api}/addDs/addOrUpdateSourceWithKerberos`,
	},
	// 测试联通性
	testCon: {
		method: 'post',
		url: `${api}/addDs/testCon`,
	},
	// 上传Kerberos测试联通性
	testConWithKerberos: {
		method: 'postForm',
		url: `${api}/addDs/testConWithKerberos`,
	},
	// 获取数据源基本详情
	detail: {
		method: 'post',
		url: `${api}/manager/detail`,
	},
	//解析kerberos文件获取principal列表
	uploadCode: {
		method: 'postForm',
		url: `${api}/addDs/getPrincipalsWithConf`,
	},
	getCreateTargetTable: {
		method: 'post',
		url: `${api}/addDs/getCreateTargetTableSql`,
	},
	createDdlTable: {
		method: 'post',
		url: `${api}/addDs/ddlCreateTable`,
	},
};
