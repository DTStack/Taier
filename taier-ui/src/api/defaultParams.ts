import { cloneDeep, isEmpty, isObject } from "lodash";

export const convertRequestParams = (template = {}, params = {}) => {
    const loop = (template: any = {}, params: any = {}) => {
        const newData: any = {}
        Object.keys(template).forEach(k => {
            if (params[k] === undefined) {
                newData[k] = undefined;
                return;
            }
            const el = template[k];
            const d = cloneDeep(params[k]);
            if (isObject(el) && !isEmpty(el) && isObject(d)) {
                newData[k] = loop(el, d);
            } else {
                newData[k] = d
            }
        })
        return newData
    }
    return loop(template, params);
}
export const restfulDataPreview = {
    bodyStr: '',
    decode: '',
    fieldDelimiter: '',
    fields: '',
    headerStr: '',
    paramStr: '',
    requestInterval: 0,
    requestMode: '',
    url: '',
    protocol: ''
}

export const taskParams = {
    additionalResourceIdList: [],
    componentVersion: '',
    computeType: 0,
    createModel: 0,
    engineType: 0,
    exeArgs: '',
    forceUpdate: false,
    graphData: [],
    id: 1,
    isDeleted: 0,
    isDirtyDataManage: true,
    lockVersion: 1,
    mainClass: '',
    name: '',
    nodePid: 1,
    operatorJsons: '',
    preSave: false,
    readWriteLockVO: {
        getLock: false,
        gmtCreate: 0,
        gmtModified: 0,
        id: 0,
        isDeleted: 0,
        lastKeepLockUserName: 1,
        lockName: '',
        modifyUserId: 1,
        projectId: 1,
        relationId: 1,
        result: 1,
        type: '',
        version: 1
    },
    resourceIdList: [],
    settingMap: {},
    side: [],
    sideParams: '',
    sink: [],
    sinkParams: '',
    source: [],
    sourceList: [],
    sourceMap: {},
    sourceParams: '',
    sqlText: '',
    strategyId: 1,
    streamTaskDirtyDataManageVO: {
        id: 0,
        linkInfo: {},
        logPrintInterval: 0,
        maxCollectFailedRows: 0,
        maxRows: 0,
        outputType: '',
        taskId: 0
    },
    scheduleConf: '',
    targetMap: {},
    taskDesc: '',
    taskId: 1,
    taskParams: '',
    taskType: 0,
    updateSource: true,
    version: 0,
    whereToEnter: 0,
    createUserId: 0,
    projectId: 0,
    tenantId: 0,
    userId: 0
}
