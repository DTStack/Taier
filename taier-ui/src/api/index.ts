import http from './http';
import req from './request';
import { TASK_TYPE_ENUM } from '@/constant';

export default {
    // 获取节点下拉
    getNodeAddressSelect(params?: any) {
        return http.post(req.GET_NODEADDRESS_SELECT, params);
    },
    // 获取类型数据源
    getTypeOriginData(params: any) {
        return http.post(req.GET_TYPE_ORIGIN_DATA, params);
    },
    listTablesBySchema(params: any) {
        return http.post(req.LIST_TABLE_BY_SCHEMA, params);
    },
    // 获取kafka topic预览数据
    getDataPreview(params: any) {
        return http.post(req.GET_DATA_PREVIEW, params);
    },
    pollPreview(params: any) {
        return http.post(req.POLL_PREVIEW, params);
    },
    // 添加或更新任务
    saveTask(params: any) {
        return http.post(req.SAVE_TASK, params);
    },
    // 获取Topic
    getTopicType(params: any) {
        return http.post(req.GET_TOPIC_TYPE, params);
    },
    getStreamTableColumn(params: { schema: string; sourceId: number; tableName: string; flinkVersion: string }) {
        return http.post(req.GET_STREAM_TABLECOLUMN, params);
    },
    // 获取源表中的时区列表
    getTimeZoneList(params?: any) {
        return http.post(req.GET_TIMEZONE_LIST, params);
    },
    // 转换向导到脚本模式
    convertToScriptMode(params: any) {
        return http.post(req.CONVERT_TO_SCRIPT_MODE, params);
    },
    checkSyntax(params: any) {
        return http.post(req.GRAMMAR_CHECK, params);
    },
    sqlFormat(params: any) {
        return http.post(req.SQL_FORMAT, params);
    },
    getTaskList(params: any) {
        return http.post(req.GET_TASK_LIST, params);
    },
    getStatusCount(params: any) {
        return http.post(req.GET_STATUS_COUNT, params);
    },
    startTask(params: any) {
        return http.post(req.START_TASK, params);
    },
    startCollectionTask(params: any) {
        return http.post(req.START_COLLECTION_TASK, params);
    },
    getTaskManagerLog(params: any) {
        return http.post(req.GET_TASK_MANAGER_LOG, params);
    },
    getJobManagerLog(params: any) {
        return http.post(req.GET_JOB_MANAGER_LOG, params);
    },
    listTaskManager(params: any) {
        return http.post(req.LIST_TASK_MANAGER, params);
    },
    getTaskLogs(params: any) {
        return http.post(req.GET_TASK_LOGS, params);
    },
    // failover 日志
    getFailoverLogsByTaskId(params: any) {
        return http.post(req.GET_TASK_FAILOVER_LOG, params);
    },
    getHistoryLog(params: any) {
        return http.post(req.GET_HISTORY_LOG, params);
    },
    isOpenCdb(params: { dataInfoId: number }) {
        return http.post(req.IS_OPEN_CDB, params);
    },
    getPDBList(params: { dataInfoId: number; searchKey?: string }) {
        return http.post(req.GET_PDB_LIST, params);
    },
    // 数据开发 - 获取启停策略列表
    getAllStrategy() {
        return http.post(req.GET_ALL_STRATEGY);
    },
    getTopicPartitionNum(params: any) {
        return http.post(req.GET_TOPIC_PARTITION_NUM, params);
    },
    getSchemaTableColumn(params: any) {
        return http.post(req.GET_SCHEMA_TABLE_COLUMN, params);
    },
    getSlotList(params: any) {
        return http.post(req.GET_SLOT_LIST, params);
    },
    getBinlogListBySource(params: any) {
        return http.post(req.GET_BINLOG_LIST_BY_SOURCE, params);
    },
    // 获取指标
    getTaskMetrics(params: { taskId: number; timespan: string; end: number; chartNames: string[] }) {
        return http.post(req.GET_TASK_METRICS, params);
    },
    getMetricValues(params: { taskId: number }) {
        // 获取所有指标
        return http.get(req.GET_METRIC_VALUES, params);
    },
    checkSourceStatus(params: { taskId: number }) {
        // 获取任务的异常数据源
        return http.post(req.CHECK_SOURCE_STATUS, params);
    },
    queryTaskMetrics(params: { taskId: number; chartName: string; timespan: string; end: number }) {
        // 查询指标数据
        return http.post(req.QUERY_TASK_METRICES, params);
    },
    getListHistory(params: any) {
        return http.post(req.GET_LIST_HISTORY, params);
    },
    listCheckPoint(params: any) {
        return http.post(req.LIST_CHECK_POINT, params);
    },
    stopTask(params: any) {
        return http.post(req.STOP_TASK, params);
    },
    getTaskJson(params: any) {
        return http.post(req.GET_TASK_JSON, params);
    },
    getTaskSqlText(params: any) {
        return http.post(req.GET_TASK_SQL_TEXT, params);
    },
    addTenant(params: any) {
        return http.post(req.ADD_TENANT, params);
    },
    switchTenant(params: { tenantId: number }) {
        return http.post(req.SWITCH_TENANT, params);
    },
    login(params: { username: string; password: string }) {
        return http.postAsFormData(req.LOGIN, params);
    },
    addCluster(params: { clusterName: string }) {
        return http.post(req.ADD_CLUSTER, params); // 新增集群
    },
    getClusterInfo(params: { clusterId: number | string }) {
        return http.get(req.GET_CLUSTER_INFO, params);
    },
    uploadResource(params: { fileName: any; componentType: any }) {
        return http.postAsFormData(req.UPLOAD_RESOURCE, params);
    },
    deleteComponent(params: { componentId: number }) {
        return http.post(req.DELETE_COMPONENT, params); // 删除组件
    },
    deleteCluster(params: { clusterId: number }) {
        return http.post(req.DELETE_CLUSTER, params);
    },
    testConnect(params: { clusterId: number; componentType: number; versionName: string }) {
        return http.post(req.TEST_CONNECT, params);
    },
    testConnects<T>(params: { clusterId: number }) {
        return http.post<T>(req.TEST_CONNECTS, params);
    },
    closeKerberos(params: { componentId: number }) {
        return http.post(req.CLOSE_KERBEROS, params);
    },
    getVersionData(params?: any) {
        return http.get(req.GET_VERSION, params);
    },
    saveComponent(params: any) {
        return http.postAsFormData(req.SAVE_COMPONENT, params);
    },
    parseKerberos(params: any) {
        return http.postAsFormData(req.PARSE_KERBEROS, params);
    },
    getClusterList(params: { currentPage: number; pageSize: number }) {
        return http.post(req.GET_CLUSTER_LIST, params);
    },
    getTenantList(params?: any) {
        return http.get(req.GET_TENANT_LIST, params);
    },
    getMetaComponent(params?: any) {
        return http.get(req.GET_META_COMPONENT, params);
    },
    // 获取存储组件列表
    getComponentStore(params: any) {
        return http.post(req.GET_COMPONENTSTORE, params);
    },
    // 上传kerberos文件
    uploadKerberos(params: any) {
        return http.postAsFormData(req.UPLOAD_KERBEROS, params);
    },
    // 更新krb5.conf文件
    updateKrb5Conf(params: { krb5Content: string }) {
        return http.post(req.UPDATE_KRB5CONF, params);
    },
    // 概览-获取集群
    getClusterDetail(params: any) {
        return http.post(req.GET_CLUSTER_DETAIL, params);
    },
    // 明细-杀死选中或者杀死全部任务
    killTasks(params: { jobResource: string; nodeAddress: string; stage: number; jobIdList?: any[] }) {
        return http.post(req.KILL_TASKS, params);
    },
    killAllTask(params: { jobResource: string; nodeAddress: string }) {
        return http.post(req.KILL_ALL_TASK, params);
    },
    stickJob(params: { jobId: string; jobResource: string }) {
        return http.post(req.JOB_STICK, params);
    },
    // 查看明细 和搜索条件
    getViewDetail(params: {
        stage: number;
        jobResource: string;
        nodeAddress?: string;
        currentPage: number;
        pageSize: number;
    }) {
        return http.post(req.GET_VIEW_DETAIL, params);
    },
    getClusterResources(params: any) {
        return http.post(req.GET_CLUSTER_RESOURCES, params);
    },
    getLoadTemplate<T>(params: any) {
        return http.post<T>(req.GET_LOADTEMPLATE, params);
    },
    getAllCluster(params?: any) {
        return http.get(req.GET_ALL_CLUSTER, params);
    },
    getEnginesByCluster(params?: any) {
        return http.get(req.GET_ENGINES_BY_CLUSTER, params);
    },
    searchTenant(params: any) {
        return http.post(req.SEARCH_TENANT, params);
    },
    bindTenant(params: any) {
        return http.post(req.BIND_TENANT, params);
    },
    switchQueue(params: any) {
        return http.post(req.SWITCH_QUEUE, params);
    },
    refreshQueue(params: { clusterName: string }) {
        return http.post(req.REFRESH_QUEUE, params);
    },
    getRetainDBList(params?: any) {
        return http.post(req.GET_RETAINDB_LIST, params);
    },
    convertToHiveColumns(params: any) {
        return http.post(req.CONVERT_TO_HIVE_COLUMNS, params);
    },
    convertDataSyncToScriptMode(params: any) {
        return http.post(req.CONVERT_SYNC_T0_SCRIPT_MODE, params);
    },
    getOfflineTaskByID<T = any>(params: any) {
        return http.post<T>(req.GET_TASK, params);
    },
    getOfflineSubTaskById<T = any>(params: any) {
        return http.post<T>(req.GET_SUB_TASK, params);
    },
    getCustomParams(params?: any) {
        return http.post(req.GET_CUSTOM_TASK_PARAMS, params);
    },
    getSyncTemplate(params: any) {
        return http.post(req.GET_SYNC_SCRIPT_TEMPLATE, params);
    },
    publishOfflineTask(params: any) {
        return http.post(req.PUBLISH_TASK, params);
    },
    getTaskTypes(params?: any) {
        return http.post(req.GET_TASK_TYPES, params);
    },
    getTableInfoByDataSource(params: any) {
        return http.post(req.GET_TABLE_INFO_BY_DATASOURCE, params);
    },
    execSQLImmediately<T>(params: any) {
        // 立即执行SQL
        return http.post<T>(req.EXEC_SQL_IMMEDIATELY, params);
    },
    stopSQLImmediately(params: any) {
        // 停止执行数据同步
        return http.post(req.STOP_SQL_IMMEDIATELY, params);
    },
    execDataSyncImmediately<T>(params: any) {
        // 立即执行数据同步
        return http.post<T>(req.EXEC_DATA_SYNC_IMMEDIATELY, params);
    },
    stopDataSyncImmediately(params: any) {
        // 停止执行SQL
        return http.post(req.STOP_DATA_SYNC_IMMEDIATELY, params);
    },
    getIncrementColumns(params: any, config?: any) {
        // 获取增量字段
        return http.post(req.GET_INCREMENT_COLUMNS, params, config);
    },
    getHivePartitions(params: any) {
        // 获取Hive分区
        return http.post(req.CHECK_HIVE_PARTITIONS, params);
    },
    selectRunLog<T>(params: any) {
        // 非数据同步接口获取日志
        return http.post<T>(req.SELECT_SQL_LOG, params);
    },
    /**
     * - 查询数据同步任务，SQL 执行结果
     * - 需要补充增量同步
     * @param {Object} params 请求参数
     * @param {Number} taskType 任务类型
     */
    selectExecResultData(params: any, taskType: any) {
        //
        const url =
            taskType && taskType === TASK_TYPE_ENUM.SYNC ? req.SELECT_DATA_SYNC_RESULT : req.SELECT_SQL_RESULT_DATA;
        return http.post(url, params);
    },
    selectExecResultDataSync(params: any) {
        // 数据同步接口获取结果表
        return http.post(req.SELECT_DATA_SYNC_RESULT, params);
    },
    selectStatus(params: any) {
        // 非数据同步接口轮训状态
        return http.post(req.SELECT_SQL_STATUS, params);
    },
    forzenTask(params: any) {
        return http.post(req.FROZEN_TASK, params);
    },
    getOfflineCatalogue(params: any) {
        return http.post(req.GET_OFFLINE_CATALOGUE, params);
    },
    addOfflineCatalogue(params: any) {
        return http.post(req.ADD_OFFLINE_CATALOGUE, params);
    },
    editOfflineCatalogue(params: any) {
        return http.post(req.EDIT_OFFLINE_CATALOGUE, params);
    },
    addOfflineResource(params: any) {
        return http.postAsFormData(req.ADD_OFFLINE_RESOURCE, params);
    },
    replaceOfflineResource(params: any) {
        return http.postAsFormData(req.REPLACE_OFFLINE_RESOURCE, params);
    },
    addOfflineTask(params: any) {
        return http.post(req.ADD_OFFLINE_TASK, params);
    },
    saveOfflineJobData(params: any) {
        return http.post(req.SAVE_OFFLINE_JOBDATA, params);
    },
    editTask(params: any) {
        return http.post(req.EDIT_TASK, params);
    },
    addOfflineFunction(params: any) {
        return http.post(req.ADD_OFFLINE_FUNCTION, params);
    },
    delOfflineTask(params: any) {
        return http.post(req.DEL_OFFLINE_TASK, params);
    },
    delOfflineFolder(params: any) {
        return http.post(req.DEL_OFFLINE_FOLDER, params);
    },
    delOfflineRes(params: any) {
        return http.post(req.DEL_OFFLINE_RES, params);
    },
    delOfflineFn(params: any) {
        return http.post(req.DEL_OFFLINE_FN, params);
    },
    getOfflineFn(params: any) {
        return http.post(req.GET_FN_DETAIL, params);
    },
    getOfflineRes(params: any) {
        return http.post(req.GET_RES_DETAIL, params);
    },
    getHBaseColumnFamily(params: any) {
        return http.post(req.GET_HBASE_COLUMN_FAMILY, params);
    },
    queryOfflineTasks(params: any) {
        return http.post(req.QUERY_TASKS, params);
    },
    getOfflineTaskLog(params: any) {
        // 获取离线任务日志
        return http.post(req.GET_TASK_LOG, params);
    },
    getOfflineTaskPeriods(params: any) {
        // 转到前后周期
        return http.post(req.GET_TASK_PERIODS, params);
    },
    queryJobs<T = any>(params: any) {
        return http.post<T>(req.QUERY_JOBS, params);
    },
    getSubJobs<T = any>(params: any) {
        return http.post<T>(req.GET_SUB_JOBS, params);
    },
    patchTaskData(params: any) {
        // 补数据
        return http.post(req.PATCH_TASK_DATA, params);
    },
    getTaskChildren<T = any>(params: any) {
        // 获取任务子节点
        return http.post<T>(req.GET_TASK_CHILDREN, params);
    },
    getRootWorkflowTask<T = any>(params: any) {
        return http.post<T>(req.GET_ROOT_WORKFLOW_TASK, params);
    },
    getFillData(params: any) {
        // 补数据搜索
        return http.post(req.GET_FILL_DATA, params);
    },
    getFillDataDetail<T = any>(params: any) {
        // 补数据详情
        return http.post<T>(req.GET_FILL_DATA_DETAIL, params);
    },

    batchStopJob(params: any) {
        // 批量停止任务
        return http.post(req.BATCH_STOP_JOBS, params);
    },
    batchRestartAndResume(params: any) {
        // 重启并恢复任务
        return http.post(req.BATCH_RESTART_AND_RESUME_JOB, params);
    },
    getJobChildren<T = any>(params: any) {
        // 获取任务子Job
        return http.post<T>(req.GET_JOB_CHILDREN, params);
    },
    getRootWorkflowJob<T = any>(params: any) {
        return http.post<T>(req.GET_ROOT_WORKFLOW_JOB, params);
    },
    queryJobStatics(params: any) {
        return http.post(req.QUERY_JOB_STATISTICS, params);
    },

    stopFillDataJobs(params: any) {
        return http.post(req.STOP_FILL_DATA_JOBS, params);
    },

    getPersonInCharge() {
        return http.post(req.USER_QUERYUSER);
    },
    /**
     * 获取工作流任务节点实例的子节点
     */
    getTaskJobWorkflowNodes(params: any) {
        return http.post(req.GET_TASK_JOB_WORKFLOW_NODES, params);
    },
    getOfflineTableList(params: any, config?: any) {
        return http.post(req.TABLE_LIST, params, config);
    },
    getOfflineTableColumn(params: any, config?: any) {
        return http.post(req.GET_TABLE_COLUMN, params, config);
    },
    getOfflineColumnForSyncopate(params: any, config?: any) {
        return http.post(req.GET_COLUMN_FOR_SYNCOPATE, params, config);
    },
    getHivePartitionsForDataSource(params: any, config?: any) {
        return http.post(req.GET_HIVE_PARTITIONS, params, config);
    },
    getDataSourcePreview(params: any) {
        return http.post(req.GET_DATA_SOURCE_PREVIEW, params);
    },
    getAllSchemas(params: any, config?: any) {
        return http.post(req.GET_ALL_SCHEMAS, params, config);
    },
    dataSourcepage(params: any) {
        return http.post(req.GET_DATA_SOURCE_PAGE, params);
    },
    typeList(params: any) {
        return http.post(req.GET_TYPE_LIST, params);
    },
    dataSourceDelete(params: any) {
        return http.post(req.DELETE_SOURCE, params);
    },
    queryDsClassifyList(params: any) {
        return http.post(req.QUERY_DATA_SOURCE_CLASSIFY_LIST, params);
    },
    queryDsTypeByClassify(params: any) {
        return http.post(req.QUERY_LIST_BY_CLASSIFY, params);
    },
    queryDsVersionByType(params: any) {
        return http.post(req.QUERY_VERSION_BY_CLASSIFY, params);
    },
    findTemplateByTypeVersion(params: any) {
        return http.post(req.QUERY_TEMPLATE_BY_VERSION, params);
    },
    addDatasource(params: any) {
        return http.post(req.ADD_DATA_SOURCE, params);
    },
    addOrUpdateSourceWithKerberos(params: any) {
        return http.postForm(req.UPLOAD_DATA_SOURCE_WITH_KERBEROS, params);
    },
    testCon(params: any) {
        return http.post(req.TEST_CONNECT_IN_DATA_SOURCE, params);
    },
    testConWithKerberos(params: any) {
        return http.postForm(req.TEST_KERBEROS_IN_DATA_SOURCE, params);
    },
    detail(params: any) {
        return http.post(req.GET_DATA_SOURCE_DETAIL, params);
    },
    uploadCode(params: any) {
        return http.postForm(req.UPLOAD_CODE, params);
    },
    getCreateTargetTable(params: any) {
        return http.post(req.GET_CREATE_TARGET_TABLE, params);
    },
    createDdlTable(params: any) {
        return http.post(req.CREATE_DDL_TABLE, params);
    },
    batchStopJobByDate(params: any) {
        // 按业务日期批量杀任务
        return http.post(req.BATCH_STOP_JOBS_BY_DATE, params);
    },
    allProductGlobalSearch(params: any) {
        return http.post(req.ALL_PRODUCT_GLOBAL_SEARCH, params);
    },
    getComponentModels() {
        return http.get(req.GET_COMPONENT_MODELS, {});
    },
    getComponentInfo(params: any) {
        return http.get(req.GET_COMPONENT_INFO, params);
    },
    getComponentSchemaConfig<T>(params: any) {
        return http.post<T>(req.GET_TENANT_COMPONENT_LIST, params);
    },
    saveComponentSchemaConfig(params: any) {
        return http.post(req.SAVE_TENANT_COMPONENT_INFO, params);
    },
    getSchemaListByComponent<T>(params: any) {
        return http.post<T>(req.GET_SCHEMA_LIST_BY_COMPONENT, params);
    },
    getResourceLocation<T = any>(params: any) {
        return http.post<T>(req.GET_RESOUCE_LOCATION, params);
    },
    validateRepeatTaskName<T = any>(params: any) {
        return http.post<T>(req.VALIDATE_REPEAT_TASK_NAME, params);
    },
    getSyncProperties<T = any>(params: any) {
        return http.post<T>(req.QUERY_SYNC_DATA_SOURCE, params);
    },
    getSupportSource<T = any>(params: any) {
        return http.post<T>(req.QUERY_SUPPORT_SOURCE, params);
    },
    getResourceByTenant(params: any) {
        return http.post(req.GET_RESOURCES_BY_TENANT, params);
    },
    getAllDataSource(params: any) {
        return http.post(req.GET_ALL_DATA_SOURCE, params);
    },
    getComponentVersionByTaskType<T = any>(params: any) {
        return http.post<T>(req.GET_COMPONENT_VERSION, params);
    },
    getFTPColumns<T = any>(params: any) {
        return http.post<T>(req.GET_FTP_COLUMNS, params);
    },
};
