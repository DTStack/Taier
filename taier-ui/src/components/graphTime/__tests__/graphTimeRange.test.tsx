import api from '@/api';
import { cleanup, fireEvent, render } from '@testing-library/react';
import { act } from 'react-dom/test-utils';
import GraphTimeRange from '../graphTimeRange';

jest.mock('@/api');

describe('Test GraphTimeRange Component', () => {
    beforeEach(() => {
        cleanup();
        document.body.innerHTML = '';

        // (api.formatTimeSpan as jest.Mock)
        // 	.mockReset()
        // 	.mockResolvedValueOnce({
        // 		code: 1,
        // 		data: {
        // 			correct: true,
        // 			formatResult: '20m',
        // 		},
        // 	})
        // 	.mockRejectedValue({
        // 		code: 1,
        // 		data: {
        // 			correct: false,
        // 		},
        // 	});
    });

    it('Should match snapshot', () => {
        expect(render(<GraphTimeRange value="10m" />).asFragment()).toMatchSnapshot();
    });

    it('Should support get range time', () => {
        const fn = jest.fn();
        const { container } = render(<GraphTimeRange value="10m" onRangeChange={fn} />);

        act(() => {
            fireEvent.click(container.querySelector('.btn-prev')!);
        });

        expect(fn).toBeCalledWith('1m');

        act(() => {
            fireEvent.click(container.querySelector('.btn-next')!);
        });

        expect(fn).toBeCalledWith('30m');
    });

    it('Should not call event handler when in the edge of range', () => {
        const fn = jest.fn();
        const { container, rerender } = render(<GraphTimeRange value="10s" onRangeChange={fn} />);

        act(() => {
            fireEvent.click(container.querySelector('.btn-prev')!);
        });

        expect(fn).not.toBeCalled();

        rerender(<GraphTimeRange value="2y" onRangeChange={fn} />);

        act(() => {
            fireEvent.click(container.querySelector('.btn-next')!);
        });

        expect(fn).not.toBeCalled();
    });

    it.skip('Should support change time by input', () => {
        const fn = jest.fn();
        const { container } = render(<GraphTimeRange value="10s" onRangeChange={fn} />);

        act(() => {
            fireEvent.change(container.querySelector('input')!, { target: { value: '20m' } });
            fireEvent.blur(container.querySelector('input')!);
        });

        expect(api.formatTimeSpan).toBeCalled();
    });
});
