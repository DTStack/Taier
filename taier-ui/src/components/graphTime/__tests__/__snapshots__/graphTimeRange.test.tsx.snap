// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test GraphTimeRange Component Should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="graph-time graph-time-range"
  >
    <button
      class="ant-btn ant-btn-default btn-prev"
      type="button"
    >
      <span
        aria-label="minus"
        class="anticon anticon-minus"
        role="img"
      >
        <svg
          aria-hidden="true"
          data-icon="minus"
          fill="currentColor"
          focusable="false"
          height="1em"
          viewBox="64 64 896 896"
          width="1em"
        >
          <path
            d="M872 474H152c-4.4 0-8 3.6-8 8v60c0 4.4 3.6 8 8 8h720c4.4 0 8-3.6 8-8v-60c0-4.4-3.6-8-8-8z"
          />
        </svg>
      </span>
    </button>
    <input
      class="ant-input"
      type="text"
      value="10m"
    />
    <button
      class="ant-btn ant-btn-default btn-next"
      type="button"
    >
      <span
        aria-label="plus"
        class="anticon anticon-plus"
        role="img"
      >
        <svg
          aria-hidden="true"
          data-icon="plus"
          fill="currentColor"
          focusable="false"
          height="1em"
          viewBox="64 64 896 896"
          width="1em"
        >
          <defs>
            <style />
          </defs>
          <path
            d="M482 152h60q8 0 8 8v704q0 8-8 8h-60q-8 0-8-8V160q0-8 8-8z"
          />
          <path
            d="M176 474h672q8 0 8 8v60q0 8-8 8H176q-8 0-8-8v-60q0-8 8-8z"
          />
        </svg>
      </span>
    </button>
  </div>
</DocumentFragment>
`;
